packages/flutter/lib/src/widgets/sensitive_content.dart

```diff
@@ -0,0 +1,289 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart' show ContentSensitivity, SensitiveContentService;
+
+import 'async.dart' show AsyncSnapshot, FutureBuilder;
+import 'container.dart';
+import 'framework.dart';
+
+/// Data structure used to track the [SensitiveContent] widgets in the
+/// widget tree.
+class ContentSensitivitySetting {
+  /// Creates a [ContentSensitivitySetting].
+  ContentSensitivitySetting();
+
+  /// The number of [SensitiveContent] widgets that have sensitivity level [ContentSensitivity.sensitive].
+  int _sensitiveWidgetCount = 0;
+
+  /// The number of [SensitiveContent] widgets that have sensitivity level [ContentSensitivity.autoSensitive].
+  int _autoSensitiveWidgetCount = 0;
+
+  /// The number of [SensitiveContent] widgets that have sensitivity level [ContentSensitivity.notSensitive].
+  int _notSensitiveWigetCount = 0;
+
+  /// Increases the count of [SensitiveContent] widgets with [sensitivityLevel] set.
+  void addWidgetWithContentSensitivity(ContentSensitivity sensitivityLevel) {
+    switch (sensitivityLevel) {
+      case ContentSensitivity.sensitive:
+        _sensitiveWidgetCount++;
+      case ContentSensitivity.autoSensitive:
+        _autoSensitiveWidgetCount++;
+      case ContentSensitivity.notSensitive:
+        _notSensitiveWigetCount++;
+    }
+  }
+
+  /// Decreases the count of [SensitiveContent] widgets with [sensitivityLevel] set.
+  void removeWidgetWithContentSensitivity(ContentSensitivity sensitivityLevel) {
+    switch (sensitivityLevel) {
+      case ContentSensitivity.sensitive:
+        _sensitiveWidgetCount--;
+      case ContentSensitivity.autoSensitive:
+        _autoSensitiveWidgetCount--;
+      case ContentSensitivity.notSensitive:
+        _notSensitiveWigetCount--;
+    }
+  }
+
+  /// Returns true if this class is currently tracking at least one [SensitiveContent] widget.
+  bool get hasWidgets =>
+      _sensitiveWidgetCount + _autoSensitiveWidgetCount + _notSensitiveWigetCount > 0;
+
+  /// Returns the most severe [ContentSensitivity] setting of the [SensitiveContent] widgets
+  /// that this setting tracks.
+  ContentSensitivity? get contentSensitivityBasedOnWidgetCounts {
+    if (_sensitiveWidgetCount > 0) {
+      return ContentSensitivity.sensitive;
+    }
+    if (_autoSensitiveWidgetCount > 0) {
+      return ContentSensitivity.autoSensitive;
+    }
+    if (_notSensitiveWigetCount > 0) {
+      return ContentSensitivity.notSensitive;
+    }
+    return null;
+  }
+}
+
+/// Host of the current content sensitivity level for the widget tree that contains
+/// some number [SensitiveContent] widgets.
+class SensitiveContentHost {
+  SensitiveContentHost._();
+
+  bool? _contentSenstivityIsSupported;
+  ContentSensitivitySetting? _contentSensitivitySetting;
```

Comment:
this can be 
```
late ContentSensitivitySetting _contentSensitivitySetting = ContentSensitivitySetting();

Reply:
Oooo I did not know about this--thank you!